### PR TITLE
only replace variables if text attribute in template

### PR DESCRIPTION
**Pull request by erohmensing**
**Proposed changes**:
- now that templates don't require `- text` (can also be `- custom` instead) we have to do an extra check to make sure we only replace variables in the `- text` attribute. Otherwise we get a KeyError which leads to a confusing error because it's captured in a try/except written for a different purpose
- should fix https://github.com/RasaHQ/rasa/issues/3745

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/nlg/template.py [hunk 1]
```diff
@@ -103,8 +103,8 @@ def _fill_template_text(
         # Getting the slot values in the template variables
         template_vars = self._template_variables(filled_slots, kwargs)
 
-        # Filling the template variables in the template
-        if template_vars:
+        # Filling the template variables in the template text
+        if template_vars and template.get("text"):
```
Comment (tabergma) [inline rasa/core/nlg/template.py]:
```suggestion
        if template_vars and "text" in template:
```

Comment (tabergma) [reply rasa/core/nlg/template.py]: I prefer this way of checking as it is more explicit.

**Comment by erohmensing** [comment]
@tmbo currently we only populate slots in the text field. Do you think we should extend this to other fields? Example use case here: https://forum.rasa.com/t/how-to-send-image-url-to-templates-dynamically/11074

Would also be a way to finally deprecate `dispatcher.utter_button_template` if you could use variables in the button field of the template

**Comment by tmbo** [comment]
Yes, but please open a separate issue for that. might be possible, but also might be to much effort.

**Comment by erohmensing** [comment]
Okay sounds good, i'll merge this one first (if the tests ever pass before master gets updated lol)

**Comment by KarthiAru** [comment]
This issue is recurring again in v1.8.2

**Comment by tmbo** [comment]
@KarthiAru can you please open a separate issue with a stacktrace?

**Pull request merged**